arrow-data/src/test/kotlin/arrow/data/TupleTest.kt

```diff
@@ -28,16 +33,21 @@ class TupleTest : UnitSpec() {
         }
 
         testLaws(
-            EqLaws.laws { Tuple2(it, it) },
-            EqLaws.laws { Tuple3(it, it, it) },
-            EqLaws.laws { Tuple4(it, it, it, it) },
-            EqLaws.laws { Tuple5(it, it, it, it, it) },
-            EqLaws.laws { Tuple6(it, it, it, it, it, it) },
-            EqLaws.laws { Tuple7(it, it, it, it, it, it, it) },
-            EqLaws.laws { Tuple8(it, it, it, it, it, it, it, it) },
-            EqLaws.laws { Tuple9(it, it, it, it, it, it, it, it, it) },
-            EqLaws.laws { Tuple10(it, it, it, it, it, it, it, it, it, it) }
+                FunctorLaws.laws(Tuple2.functor(), { Tuple2.pure(it) }, Eq.any()),
```

Comment:
There is no need to repeat the laws for every typeclass in a hierarchy. MonadLaws internally call to Applicative and Functor laws. Traverse laws call to Foldable laws too.